std/algorithm/searching.d

```diff
@@ -1406,7 +1405,15 @@ if (isInputRange!Range && !isInfinite!Range &&
             }
         }
     }
-    return extremeElement;
+    // Rebindable is an alias to T for arrays
+    static if (is(typeof(extremeElement) == T[], T))
+    {
+        return extremeElement;
+    }
+    else
+    {
+        return extremeElement.get;
+    }
```

Comment:
Can we use `extremeElement.get` in both cases?

Reply:
No: see #8735; sometimes `Rebindable!T` is `T`.

Reply:
To phrase my question differently:

Can we add a `get` template alias member to the `is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArray!T` version of `Rebindable`, to allow it to play nice with generic code? My guess is that the opposite was undesirable for some reason - having `alias get this` in the non-ref case.

Reply:
Another option is to add a `package`-private `rebindableGet` free function that could work genetically in both cases.